src/MoBi.Presentation/Tasks/SimulationRunner.cs

```diff
@@ -97,44 +113,95 @@ private bool settingsRequired(IMoBiSimulation simulation, bool defineSettings)
+               _context.PublishEvent(new SimulationsRunCanceledEvent());
+            }
+         }
+         else
+         {
+            foreach (var cts in _cancellationTokenSources.Values.ToList())
+            {
+               cts.Cancel();
+               cts.Dispose();
+            }
+
+            _cancellationTokenSources.Clear();
+            _context.PublishEvent(new SimulationsRunCanceledEvent());
+         }
       }
 
-      private void startSimulationRun(IMoBiSimulation simulation)
+      private async Task startSimulationRunAsync(IMoBiSimulation simulation)
       {
-         _simulation = simulation;
-         _context.PublishEvent(new SimulationRunStartedEvent());
+         if (simulation.IsRunning)
+            return;
+
+         var cts = new CancellationTokenSource();
+         _cancellationTokenSources[simulation] = cts;
+         _context.PublishEvent(new SimulationRunStartedEvent(simulation));
+         setSimulationRunning(simulation, true);
          _context.PublishEvent(new ProgressInitEvent(100, AppConstants.SimulationRun));
          _simModelManager = _simModelManagerFactory.Create();
-
+         
          try
          {
             addEvents();
             updatePersistableFor(simulation);
-            var simulationRunResults = _simModelManager.RunSimulation(_simulation);
-            _simulation.HasChanged = true;
+
+            var simulationRunResults = await _simModelManager.RunSimulationAsync(simulation, cts.Token);
+
+            simulation.HasChanged = true;
             showWarningsIfAny(simulationRunResults);
 
             if (simulationRunResults.Success)
             {
                var results = simulationRunResults.Results;
                results.Name = getNewRepositoryName();
                _displayUnitUpdater.UpdateDisplayUnitsIn(results);
-               copyResultsToSimulation(simulationRunResults, _simulation);
+               copyResultsToSimulation(simulationRunResults, simulation);
             }
 
             addCommand(getSimulationResultLabel(simulationRunResults));
          }
          finally
          {
+            if (_cancellationTokenSources.ContainsKey(simulation))
+            {
+               _cancellationTokenSources[simulation].Dispose();
+               _cancellationTokenSources.Remove(simulation);
+            }
             removeEvents();
-            _context.PublishEvent(new SimulationRunFinishedEvent(_simulation));
-            _simulation = null;
+            _context.PublishEvent(new SimulationRunFinishedEvent(simulation));
+            setSimulationRunning(simulation, false);
          }
       }
 
+      private void setSimulationRunning(IMoBiSimulation simulation, bool isRunning)
+      {
+         if (isRunning)
+         {
+            if (!_originalSimulationNames.ContainsKey(simulation))
+               _originalSimulationNames[simulation] = simulation.Name;
+
+            if (!simulation.Name.EndsWith(" (Running)"))
```

Comment:
NO! Please le'ts npot rename the simulation to run them. Why

Reply:
This is presentation. So if you want to show in the tree, it could be done by handling an event (simulariion, running or finished etc.). If something goes wrong for some reason, this could lead to serious inconsistency